exchanges/binance/binance_test.go

```diff
@@ -36,14 +36,1035 @@ func setFeeBuilder() *exchange.FeeBuilder {
 	}
 }
 
+func TestUpdateTicker(t *testing.T) {
+	t.Parallel()
+	cp, err := currency.NewPairFromString("BTCUSD_201225")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = b.UpdateTicker(cp, asset.CoinMarginedFutures)
+	if err != nil {
+		t.Error(err)
+	}
+	cp2, err := currency.NewPairFromString("BTCUSDT")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = b.UpdateTicker(cp2, asset.USDTMarginedFutures)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdateOrderbook(t *testing.T) {
+	t.Parallel()
+	cp, err := currency.NewPairFromString("BTCUSDT")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = b.UpdateOrderbook(cp, asset.USDTMarginedFutures)
+	if err != nil {
+		t.Error(err)
+	}
+	cp2, err := currency.NewPairFromString("BTCUSD_PERP")
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = b.UpdateOrderbook(cp2, asset.CoinMarginedFutures)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+// USDT Margined Futures
+
+func TestUExchangeInfo(t *testing.T) {
+	t.Parallel()
+	_, err := b.UExchangeInfo()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUFuturesOrderbook(t *testing.T) {
+	t.Parallel()
+	_, err := b.UFuturesOrderbook("BTCUSDT", 1000)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestURecentTrades(t *testing.T) {
+	t.Parallel()
+	_, err := b.URecentTrades("BTCUSDT", "", 5)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUHistoricalTrades(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
```

Comment:
The mocked data for `/fapi/v1/historicalTrades`  is `{"code":-2014,"msg":"API-key format invalid."}`
For this function, it results in an error about cannot unmarshall
```
Binance request path: https://127.0.0.1:57237/fapi/v1/historicalTrades?limit=5&symbol=BTCUSDT
Binance request type: GET
HTTP status: 200 OK, Code: 200
Binance raw response: {"code":-2014,"msg":"API-key format invalid."}
    binance_test.go:113: json: cannot unmarshal object into Go value of type []binance.UPublicTradesData
```